build/common.sh

```diff
@@ -272,7 +272,7 @@ function kube::build::docker_image_exists() {
 
   # We cannot just specify the IMAGE here as `docker images` doesn't behave as
   # expected.  See: https://github.com/docker/docker/issues/8048
-  "${DOCKER[@]}" images | grep -Eq "^${1}\s+${2}\s+"
+  "${DOCKER[@]}" images | grep -Eq "${1}\s+${2}\s+"
```

Comment:
I do not see this behavior.  I'm wary of loosening the regex without understanding why..

```console
thockin@freakshow kubernetes master /$ docker version
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 786b29d
OS/Arch (client): linux/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 786b29d
OS/Arch (server): linux/amd64

thockin@freakshow kubernetes master /$ docker pull busybox
latest: Pulling from busybox

cf2616975b4a: Already exists 
6ce2e90b0bc7: Already exists 
8c2e06607696: Already exists 
Digest: sha256:38a203e1986cf79639cfb9b2e1d6e773de84002feea2d4eb006b52004ee8502d
Status: Image is up to date for busybox:latest

thockin@freakshow kubernetes master /$ docker images | grep busybox
busybox                                     latest              8c2e06607696        3 months ago        2.433 MB

thockin@freakshow kubernetes master /$ docker rmi busybox
Untagged: busybox:latest
Deleted: 8c2e06607696bd4afb3d03b687e361cc43cf8ec1a4a725bc96e39f05ba97dd55
Deleted: 6ce2e90b0bc7224de3db1f0d646fe8e2c4dd37f1793928287f6074bc451a57ea
Deleted: cf2616975b4a3cba083ca99bc3f0bf25f5f528c3c52be1596b30f60b0b1c37ff

thockin@freakshow kubernetes master /$ docker pull docker.io/busybox
latest: Pulling from busybox

cf2616975b4a: Pull complete 
6ce2e90b0bc7: Pull complete 
8c2e06607696: Already exists 
busybox:latest: The image you are pulling has been verified. Important: image verification is a tech preview feature and should not be relied on to provide security.

Digest: sha256:38a203e1986cf79639cfb9b2e1d6e773de84002feea2d4eb006b52004ee8502d
Status: Downloaded newer image for busybox:latest

thockin@freakshow kubernetes master /$ docker images | grep busybox
busybox                                     latest              8c2e06607696        3 months ago        2.433 MB
```


Maybe someone from Docker can explain why you see a `docker.io/` prefix on images but I don't?  @jfrazzelle for a consult..